packages/gatsby-transformer-mdx-introspection/gatsby-node.js

```diff
@@ -0,0 +1,88 @@
+const remark = require('remark');
+const mdx = require('remark-mdx');
+const generateComponentNodeObject = require('./src/generate-component-node-object');
+
+exports.sourceNodes = ({ actions }) => {
+  actions.createTypes(`
+    type ComponentInMdx implements Node @dontInfer {
+      component: String!
+      attributes: [Attribute!]
+    }
+
+    type Attribute {
+      name: String!
+      value: String
+    }
+  `);
+};
+
+exports.onCreateNode = ({
+  node,
+  actions,
+  createNodeId,
+  createContentDigest,
+}) => {
+  if (node.internal.type === 'Mdx') {
+    introspectMdx({
+      node,
+      createNodeId,
+      createContentDigest,
+      actions,
+    });
+  }
+};
+
+async function introspectMdx({
+  node,
+  createNodeId,
+  createContentDigest,
+  actions,
+}) {
+  await remark()
+    .use(mdx)
+    .use(() => tree => {
+      const { children } = tree;
```

Comment:
Nitpick: unnecessary destructure. Simply use `tree.children.forEach`

Reply:
right, that's a leftover from an earlier iteration that needed more data from the tree.